src/main.cpp

```diff
@@ -2046,18 +2040,26 @@ static int64_t nTimeTotal = 0;
 
 static bool DidBlockTriggerSizeFork(const CBlock &block, const CBlockIndex *pindex, const CChainParams &chainparams)
 {
-    if (pblocktree->ForkBitActivated(FORK_BIT_2MB) != uint256())
+    if (pblocktree->IsSizeRemovalForkActivated())
         return false; // Already active
-    if (block.nTime > chainparams.GetConsensus().SizeForkExpiration()) {
-        // 2MB vote failed: this code is obsolete
-        strMiscWarning = _("Warning: This version is obsolete; upgrade required!");
-        AlertNotify(strMiscWarning, true);
-        uiInterface.NotifyAlertChanged();
+    if (block.nTime > chainparams.GetConsensus().nSizeForkExpiration)
+        return false; // won't become active anymore.
+    if ((block.nVersion & CBlockHeader::SIZE_FORK_BIT) != CBlockHeader::SIZE_FORK_BIT)
         return false;
+
+    int found = 0;
+    const Consensus::Params &params = chainparams.GetConsensus();
+    const CBlockIndex *current = pindex;
+    for (int i = 0; i < params.nMajorityWindow && current; ++i)
+    {
+        if ((current->nVersion & CBlockHeader::SIZE_FORK_BIT) == CBlockHeader::SIZE_FORK_BIT) {
+            if (++found > params.nActivateSizeForkMajority)
```

Comment:
I thought the same at first but that `++found` is a little tricky. Does the C++ standard say anything about how sequence points apply to pre-increment operators? Does the short circuit ensure `++` does not happen?